src/device/mod.rs

```diff
@@ -1997,6 +1998,32 @@ impl TaskRuntime for Runtime {
             },
 
             Some(mesh_event) = self.event_listeners.wg_event_subscriber.recv() => {
+                let public_key = mesh_event.peer.public_key;
+
+                if let Some(mesh_entities) = self.entities.meshnet.as_ref() {
+                    if let Some(proxy_endpoints) = mesh_entities.proxy.get_endpoint_map().await.ok().and_then(|proxy_map| proxy_map.get(&public_key).cloned()) {
```

Comment:
This `.cloned()` could've been avoided with either storing `proxy_map` in a variable or simply using `remove()` instead of `get()` since the map itself is no longer used